src/main/java/bio/terra/cli/context/resources/GcsBucketLifecycle.java

```diff
@@ -0,0 +1,163 @@
+package bio.terra.cli.context.resources;
+
+import bio.terra.workspace.model.GcpGcsBucketLifecycleRule;
+import bio.terra.workspace.model.GcpGcsBucketLifecycleRuleAction;
+import bio.terra.workspace.model.GcpGcsBucketLifecycleRuleActionType;
+import bio.terra.workspace.model.GcpGcsBucketLifecycleRuleCondition;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This POJO class specifies a list of GCS bucket lifecycle rules. Its structure mimics the one used
+ * by the `gsutil lifecycle` command.
+ *
+ * <p>Command ref: https://cloud.google.com/storage/docs/gsutil/commands/lifecycle
+ *
+ * <p>GCS bucket JSON format ref: https://cloud.google.com/storage/docs/json_api/v1/buckets#resource
+ *
+ * <p>This class is intended for passing throughout the CLI codebase, and also for exposing to users
+ * so that they can specify the lifecycle rules in a JSON file and pass that to the CLI.
+ */
+@SuppressFBWarnings(
+    value = {"UWF_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD", "NP_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD"},
+    justification = "This POJO class is used for easy serialization to JSON using Jackson.")
+public class GcsBucketLifecycle {
+  public List<Rule> rule = new ArrayList<>();
+
+  public static class Rule {
+    public Action action;
+    public Condition condition;
+  }
+
+  public static class Action {
+    public ActionType type;
+    public GcsStorageClass storageClass;
+  }
+
+  public static class Condition {
+    public Integer age;
+    public LocalDate createdBefore;
```

Comment:
With this option, I was trying to mimic the `gsutil lifecycle` command as closely as possible. That command takes dates in YYYY-MM-DD format, and I believe assumes that the time is midnight and the timezone is UTC.
Reference: https://cloud.google.com/storage/docs/json_api/v1/buckets#lifecycle)
The note about midnight/UTC is only included in the description for the `createdBefore` field, but I assumed it applies to the others since the format is specified in the same way.